test/integration/stats/applications_test.rb

```diff
@@ -15,12 +17,15 @@ def setup
   def test_show
     get admin_buyers_stats_application_path(id: @application.id)
     assert_response :success
+  end
 
+  def test_show_without_access_to_all_services
     User.any_instance.expects(:has_access_to_all_services?).returns(false).at_least_once
     get admin_buyers_stats_application_path(id: @application.id)
     assert_response :forbidden
+  end
 
-    User.any_instance.expects(:member_permission_service_ids).returns([@service.id]).at_least_once
+  def test_show_with_member_permissions
```

Comment:
This test seems to be a duplicate of `test_show` now. Probably issue comes from splitting the test cases. Probably this can help but I didn't test.
```suggestion
  def test_show_with_member_permissions
    User.any_instance.expects(:has_access_to_all_services?).returns(false).at_least_once
    User.any_instance.expects(:member_permission_service_ids).returns([@service.id]).at_least_once
```

Or maybe merge back the last two test cases.